### DISCUSSION TITLE
Maxi 10 onduleur

### DISCUSSION THREAD

**Discussion by jacques3629** [tbnobody/OpenDTU#1477] [General]
Bonjour 

je sais que la question a déjà été abordée 

Déjà félicitation au développeur, un ensemble qui ne coute pas très cher, simple à mettre en œuvre 
actuellement, j'ai 16 micros onduleur, je tourne avec 2 opendtu  ça fonctionne bien 
aucun risque ?
j'ai cru comprendre que le max onduleur était de 10 onduleurs
le problème est-il lié à la mémoire de l'esp32 ?
À l'avenir, si une nouvelle puce mise sur le marché avec plus de mémoire, prévoyez-vous d'adapter le PRG 

merci pour votre travail

traduction google

Good morning

I know this question has already been addressed

Already congratulations to the developer, a set which does not cost very expensive, simple to implement
currently, I have 16 micro inverters, I run with 2 opendtu it works well
no risk ?
I understood that the max inverter was 10 inverters
is the problem related to the esp32 memory?
In the future, if a new chip comes to market with more memory, do you plan to adapt the GWP

thank you for your work

**Comment by stefan123t** [comment]
Salut @jacques3629 
OpenDTU has a fixed limit of max 10 inverters. 

The reason as you hinted are large data structures especially for Inverters with like 4 channels and another memory eater are the Alarm Data structures which can be up to 14x2 Events per inverter.

Which inverters do you have in specific ?

You can either try to adapt the code on your own risk, after all it is OpenSource and try if it works for your situation.

We have one user with 16 Inverters (all HM-400 single channel only) which therefor is the current default limit for AhoyDTU the sister project of OpenDTU. So you may try the latest version of AhoyDTU but keep in mind that the MQTT Topic structure and REST API are not exactly the same due to historic reasons.

So depending on your use case and third-party usage / post processing of data from your inverters this may be a show stopper for you.

**Comment by jacques3629** [comment]
Ho là, je n'ai pas votre talent pour me plonger dans la modification d'un code, j'ai déjà du mal à suivre un tutoriel   :))
j'ai 7 HM 400 + 5 HM 700+ 1 HMS 1000   total 13 onduleur   pour 19 panneaux  
 opendtu avec 2 boitiers fonctionne bien 
mais comme je suis curieux, je vais certainement essayer aussi AhoyDTU

merci pour votre travail 
Cordialement    


Hey there, I don't have your talent to delve into modifying code, I already have difficulty following a tutorial :))
I have 7 HM 400 + 5 HM 700+ 1 HMS 1000 total 13 inverters for 19 panels
  opendtu with 2 boxes works well
but since I'm curious, I'll definitely try AhoyDTU too

thank you for your work
Sincerely

**Comment by garg-art2002** [comment]
Hi Folks:  I realize the hardware is cheap but it means using different IP addresses (which may not be an issue if viewed through the Home Assistant).

I have 30 single module inverters.  I wonder if the change is a big one.  Also, I would like to make a suggestion to limit the number of DC modules. @stephan123t commented about 4 channels in the comment above. I have only one channel, I wonder if there is a solution. AhoyDTU would still force two ip addresses. Hence looking for a way to suggest to the developers.

**Reply by stefan123t** [reply]
@garg-art2002 this is in fact the case for the Hoymiles DTU Pro which has a limit of 99 or 100 PV modules *afair*.
Looking at it from a channel / PV-module perspective makes the memory structures a lot simpler, as they are all simply the same.
You may have an additional one for the total sum of the inverter, hence you can manage and monitor up to 25/20x 4-in-1 channel inverters or 50/33x 2-in-1 channel inverters or a whopping 100x 1-in-1 channel inverters.
